Datasets/MNIST/MNIST.swift

```diff
@@ -0,0 +1,104 @@
+// Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import TensorFlow
+
+public struct MNIST {
+    public let trainImages: Tensor<Float>
+    public let trainLabels: Tensor<Int32>
+    public let testImages: Tensor<Float>
+    public let testLabels: Tensor<Int32>
+
+    public let trainingSize: Int
+    public let testSize: Int
+
+    let batchSize: Int
```

Comment:
I believe this should also be public, because:
1. The initializer takes this as an argument. Since the user already explicitly provides it, it's weird to hide it from the user.
2. Some pilot users like Jeremy Howard have strongly recommend making model properties be public, which is considered important for research flexibility.

```suggestion
    public let batchSize: Int
```

Reply:
I'll keep those guidelines in mind, as well.